Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/chan55/Chan55.java

```diff
@@ -0,0 +1,308 @@
+import com.github.adamantcheese.chan.core.site.SiteIcon;
+import com.github.adamantcheese.chan.core.site.common.CommonReplyHttpCall;
+import com.github.adamantcheese.chan.core.site.common.CommonSite;
+import com.github.adamantcheese.chan.core.site.common.MultipartHttpCall;
+import com.github.adamantcheese.chan.core.site.common.vichan.VichanActions;
+import com.github.adamantcheese.chan.core.site.common.vichan.VichanApi;
+import com.github.adamantcheese.chan.core.site.common.vichan.VichanCommentParser;
+import com.github.adamantcheese.chan.core.site.common.vichan.VichanEndpoints;
+import com.github.adamantcheese.chan.core.site.http.DeleteRequest;
+import com.github.adamantcheese.chan.core.site.http.HttpCall;
+import com.github.adamantcheese.chan.core.site.http.ProgressRequestBody;
+import com.github.adamantcheese.chan.core.site.http.Reply;
+import com.github.adamantcheese.chan.core.site.http.ReplyResponse;
+import com.github.adamantcheese.chan.core.site.sites.dvach.Dvach;
+import com.github.adamantcheese.chan.core.site.sites.dvach.DvachReplyCall;
+import com.github.adamantcheese.chan.ui.layout.ReplyLayout;
+import com.github.adamantcheese.chan.utils.Logger;
+
+import org.jsoup.Jsoup;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.HttpUrl;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class Chan55 extends CommonSite {
+    public static final CommonSiteUrlHandler URL_HANDLER = new CommonSiteUrlHandler() {
+        @Override
+        public Class<? extends Site> getSiteClass() {
+            return Chan55.class;
+        }
+
+        @Override
+        public HttpUrl getUrl() {
+            return HttpUrl.parse("https://55chan.org/");
+        }
+
+        @Override
+        public String[] getNames() {
+            return new String[]{"55chan"};
+        }
+
+        @Override
+        public String desktopUrl(Loadable loadable, @Nullable Post post) {
+            if (loadable.isCatalogMode()) {
+                return getUrl().newBuilder().addPathSegment(loadable.boardCode).toString();
+            } else if (loadable.isThreadMode()) {
+                return getUrl().newBuilder()
+                        .addPathSegment(loadable.boardCode).addPathSegment("res")
+                        .addPathSegment(loadable.no + ".html")
+                        .toString();
+            } else {
+                return getUrl().toString();
+            }
+        }
+    };
+
+    private static final String TAG = "Chan55";
+    private static final String BASE_URL = "https://55chan.org";
+    private final Random secureRandom = new SecureRandom();
+
+    @Override
+    public void setup() {
+        setName("55chan");
+        setIcon(SiteIcon.fromFavicon(HttpUrl.parse("https://55chan.org/static/favicon.ico?v=2")));
```

Comment:
I did notice that the favicon is a GIF and not an actual ICO file. If it renders as a static image that's fine, but be aware of that if you wanted it to be animated.